garage/tf/policies/gaussian_mlp_policy_with_model.py

```diff
@@ -3,10 +3,11 @@
 import tensorflow as tf
 
 from garage.tf.models.gaussian_mlp_model import GaussianMLPModel
+from garage.tf.policies.base2 import StochasticPolicy2
 from garage.tf.spaces import Box
 
 
-class GaussianMLPPolicyWithModel:
+class GaussianMLPPolicyWithModel(StochasticPolicy2):
```

Comment:
Can we just abandon GaussianMLPPolicyWithModel for good? Here it inherits StochasticPolicy2. And GaussianMLPPolicyWithModel2 inherits from StochasticPolicy2, too? It is really confusing to me which to inherit from which.

Reply:
It will take some effort to make the existing `GaussianMLPPolicyWithModel2` work with algorithm because the API is very different.
`GaussianMLPPolicyWithModel2` does not inherit from `StochasticPolicy2`. The reason why `GaussianMLPPolicyWithModel` does is that it needs to comply with the algorithm API.

Reply:
Yes it needs take the effort to refactor the algorithm, that’s what we have to do eventually. But it would make more sense if you delete GaussianMLPPolicyWithModel, have another ppoWithModel to compatible with the current GaussianMLPPolicyWithModel2.

Reply:
IMO if you have some xx and xx2 naming, model should inherit policy, model2 should inherit policy2. But right now it doesn’t do this. It makes me confused.

Reply:
That makes sense. If everyone is ok with that, I will remove `GaussianMLPPolicyWithModel` and make a ppoWithModel.

Reply:
and I will name everything with model with suffix `WithModel`.

Reply:
with original `ppo` and `GaussianMLPPolicyWithModel`, most of the changes are about *How GaussianMLPolicy* will look like. I didn't expect to introduce `ppoWithModel` in this PR. That's why I want to make minimal changes to `ppo` but to show how `GaussianMLPolicy` can be used with an algorithm.

Reply:
If that makes more confusion, I agree it's better to move on and make `ppoWithModel`.

Reply:
With your latest change, would PPO work with other policies that is not implemented with Model? I.e. GaussianLSTMPolicy.

Reply:
I’m ok if we delete GaussianMLPPolicyWithModel and only add GaussianMLPPoliyWithModel2 in this pr. Add PPOWithModel seems to be the scope of another pr

Reply:
I can make `ppoWithModel` in another PR, so we won't be seeing test which uses `GaussianMLPPolicyWithModel` in an algorithm, in this PR.

Reply:
I agree with @ahtsan to make minimal changes to algorithms in this pr but I don't agree with writing `PPOWithModel`. We should just refactor algorithms when we are refactoring for using tf distributions. 

I will suggest the following: refactor policies/baselines with models and keep garage distribution at this point, then replace all distribution with tf distribution in the next pr.

I don't understand why model should inherit policy. @CatherineSue